keel-api/src/test/kotlin/com/netflix/spinnaker/keel/rest/ApplicationEventTests.kt

```diff
@@ -0,0 +1,66 @@
+package com.netflix.spinnaker.keel.rest
+
+import com.netflix.spinnaker.keel.KeelApplication
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.MOCK
+import org.springframework.context.ApplicationEvent
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.ApplicationListener
+import org.springframework.stereotype.Component
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import strikt.api.expectThat
+import strikt.assertions.isNotEqualTo
+import strikt.assertions.startsWith
+import java.time.Duration
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(
+  classes = [KeelApplication::class, ThreadCapturingEventListener::class],
+  webEnvironment = MOCK
+)
+internal class ApplicationEventTests {
+
+  @Autowired
+  lateinit var publisher: ApplicationEventPublisher
+
+  @Autowired
+  lateinit var listener: ThreadCapturingEventListener
+
+  @Test
+  fun `events are dispatched on a different thread`() {
+    val testThread = Thread.currentThread()
+
+    publisher.publishEvent(TestEvent(this))
+
+    val eventThread = listener.awaitInvoked(Duration.ofMillis(100))
+
+    expectThat(eventThread)
+      .isNotEqualTo(testThread)
+      .get { name }
+      .startsWith("event-pool-")
```

Comment:
Thought it was worth testing the events actually do get handled on another thread since it's dependent on bean-naming magic.